src/coreclr/src/jit/importer.cpp

```diff
@@ -8362,8 +8363,15 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
 
 DONE:
 
-    if (tailCall)
+    // Final importer checks for calls flagged as tail calls.
+    //
+    if (tailCallFlags != 0)
     {
+        const bool isExplicitTailCall = (tailCallFlags & PREFIX_TAILCALL_EXPLICIT) != 0;
+        const bool isImplicitTailCall = (tailCallFlags & PREFIX_TAILCALL_IMPLICIT) != 0;
+
+        assert(isExplicitTailCall || isImplicitTailCall);
```

Comment:
Shouldn't this assert be `isExplicitTailCal != isImplicitTailCall`? (Or they can be both?)

Reply:
It can be both

Reply:
No, can't be both. Let me update this.